counterparty-core/counterpartycore/server.py

```diff
@@ -782,6 +783,16 @@ def start_all(args):
         backend.addrindexrs.stop()
         log.shutdown()
         fetcher.stop()
+        try:
+            database.check_wal_file()
+        except exceptions.WALFileFoundError:
+            logger.error(
+                "Database WAL file detected. To ensure no data corruption has occurred, run `counterpary-server check-db`."
+            )
+        except exceptions.DatabaseError:
+            logger.error(
+                "Database in use by another process. Please run `counterpary-server check-db` to verify its integrity."
```

Comment:
Does this exception get thrown only when the DB is already open?

If it's already opened by another running process, how will running `check-db` help?

Reply:
yep!
to check if the other process didn't corrupt the db ?

Reply:
Hm, I don't follow. Why would the other process corrupt the DB? If there's another process using the DB (e.g. the `sqlite` CLI), seems like the correct soln is just to close it and try again.

Reply:
we can't close a connection opened by another process (`sqlite3` for example). But I agree the message is not really relevant. Maybe juste a warning "another process is using the database, unable to close it correctly." ?

Reply:
yep! that's what I meant! just `"Database in use by another process. Please close it and try again."`

Reply:
oh! you mean refuse to terminate `counterparty-server` until the other process is closed? hmm.. this can cause problems with `docker stop` which at one time or another will send a SIGKILL in any case.. It's generally annoying to have a program that refuses to quit...

Reply:
ah right this is just for shutdown! let's make the error: "Database is in use by another process and was unable to be closed correctly."

Reply:
done